go/vt/mysqlctl/backup.go

```diff
@@ -343,8 +342,8 @@ func Restore(ctx context.Context, params RestoreParams) (*BackupManifest, error)
 	// This is safe, since we're restarting MySQL after the restore anyway
 	params.Logger.Infof("Restore: disabling super_read_only")
 	if err := params.Mysqld.SetSuperReadOnly(false); err != nil {
-		if strings.Contains(err.Error(), Error1193) {
-			params.Logger.Warningf("Restore: server does not know about super_read_only; maybe MariaDB? Continuing anyway.")
+		if strings.Contains(err.Error(), strconv.Itoa(mysql.ERUnknownSystemVariable)) {
```

Comment:
This feels a little brittle to me, but it's not new in this PR. This should be an INVALID_ARGUMENT vitess error right? https://github.com/vitessio/vitess/blob/f238c0bb9ab5757b2f9d1771081e899ea06f2df6/go/vt/vttablet/tabletserver/tabletserver.go#L1575-L1592

So can we leverage this here? https://pkg.go.dev/errors#Is 

```
if err.Is(err, mysql.ERUnknownSystemVariable) {
```
Or if that doesn't work for some reason:
```
if err.Is(err, vtrpcpb.Code_INVALID_ARGUMENT) {
```

This is non-blocking though, as we were already doing the substring search.

Reply:
This is a good idea. We should combine the Code + substring. I'll make the change before I merge.

Reply:
I forgot that we can't do this yet. See https://github.com/vitessio/vitess/pull/9240#discussion_r754662749
Here's where we lose the type information
https://github.com/vitessio/vitess/blob/d74de0c575396dcc0fdc4e4fc9dc00f799638652/go/vt/mysqlctl/query.go#L84
I opened #9311 for this. Once that is done, we can clean up error handling in these files.